ex1.c

```diff
@@ -157,7 +172,7 @@ int main( int argc, char* argv[] )
 	dsize[1] = global_size[1]/psize[1] + 2;
 	
 	// create a 2D Cartesian MPI communicator & get our coordinate (rank) in it
-	int cart_period[2] = { 0, 0 };
+	int cart_period[2] = { 1, 1 };
 	MPI_Comm cart_comm; MPI_Cart_create(main_comm, 2, psize, cart_period, 1, &cart_comm);
```

Comment:
`assert(psize[1]*psize[0] == psize_1d);` is called before the creation of `cart_comm`, `psize` is updated here.

Reply:
psize is read from the YAML file before:

// load the parallelism configuration
PC_int(PC_get(conf, ".parallelism.height"), &longval); psize[0] = longval;
PC_int(PC_get(conf, ".parallelism.width" ), &longval); psize[1] = longval;